File: Tabnine/src/main/java/com/tabnine/statusBar/StatusBarPopupListener.java
package com.tabnine.statusBar;

import com.intellij.openapi.ui.popup.JBPopupListener;
import com.intellij.openapi.ui.popup.LightweightWindowEvent;
import com.tabnine.binary.BinaryRequestFacade;
import com.tabnine.binary.requests.statusBar.StatusBarInteractionRequest;
import com.tabnine.general.DependencyContainer;
import org.jetbrains.annotations.NotNull;

public class StatusBarPopupListener implements JBPopupListener {
  private final BinaryRequestFacade binaryRequestFacade =
      DependencyContainer.instanceOfBinaryRequestFacade();

  @Override
  public void beforeShown(@NotNull LightweightWindowEvent event) {
    binaryRequestFacade.executeRequest(new StatusBarInteractionRequest());
  }
}

File: Tabnine/src/main/java/com/tabnine/statusBar/StatusBarPopupListener.kt
package com.tabnine.statusBar

import com.intellij.openapi.components.service
import com.intellij.openapi.ui.popup.JBPopupListener
import com.intellij.openapi.ui.popup.LightweightWindowEvent
import com.tabnineCommon.binary.BinaryRequestFacade
import com.tabnineCommon.binary.requests.statusBar.StatusBarInteractionRequest
import com.tabnineCommon.general.IProviderOfThings

class StatusBarPopupListener : JBPopupListener {
    private val binaryRequestFacade: BinaryRequestFacade = service<IProviderOfThings>().binaryRequestFacade
    override fun beforeShown(event: LightweightWindowEvent) {
        binaryRequestFacade.executeRequest(StatusBarInteractionRequest())
    }
}
